Add SA executemany compiled cache support

katajakasa opened the pull request:
Add compiled_cache support for SAConnection.executemany. Add tests for INSERT and UPDATE cases.

terrycain commented on aiomysql/sa/connection.py:
```diff
@@ -103,7 +103,18 @@ def _base_params(self, query, dp, compiled, is_update):
+                compiled = self._compiled_cache.get(key)
+                if not compiled:
+                    compiled = query.compile(dialect=self._dialect)
+                    dp_checks = [dp and dp.keys() == compiled.params.keys()
```
Perhaps a comment here what this list comprehension does, for a non sqlalchemy user, I have no clue what this does.

katajakasa replied:
Resolving this conversation due to the checks being removed :)

katajakasa commented:
Okay, so I dropped the extra checks. Turns out those were not actually needed plus they were a perf hit when adding larger amounts of rows. Also added a testcase for DELETE :)

terrycain commented:
This good to go now?

katajakasa commented:
@terrycain Yes, this is good to go.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/aio-libs/aiomysql?pullRequest=376) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/aio-libs/aiomysql?pullRequest=376) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/aio-libs/aiomysql?pullRequest=376) it.</sub>
